scout: clarify data handling for local analysis (#17907)
docker/docs: scout/data-handling.md

displayed in the user interfaces where Docker Scout data is surfaced, such as
the Docker Scout Dashboard and in Docker Desktop.

> For images analyzed locally on a developer's machine, Docker Scout only

Reviewer: ```suggestion
For images analyzed locally on your machine, Docker Scout only
```
?
